pycbc/workflow/core.py

```diff
@@ -181,16 +181,18 @@ def __init__(self, cp, name, ifos=None, out_dir=None, tags=None,
 
         if exe_url.scheme in ['', 'file']:
             # NOTE: There could be a case where the exe is available at a
-            #       remote site, but not on the submit host. We could work to
-            #       allow this if it ever becomes a viable use-case. Some other
-            #       places (e.g. versioning) would have to be edited as well.
+            #       remote site, but not on the submit host. Currently allowed
+            #       for the OSG site, versioning will not work as planned if
+            #       we can't see the executable (can we perhaps run versioning
+            #       including singularity??)
 
             # Check that executables at file urls
             #  on the local site exist
-            if os.path.isfile(exe_url.path) is False:
-                raise TypeError("Failed to find %s executable "
-                                "at %s on site %s" % (name, exe_path,
-                                exe_site))
+            if not exe_site in ['osg']:
```

Comment:
Is there some way we can avoid some of the site-specific places within this level of code? I'm not sure this is the best alternative, but one option would be to have a executeable config flag that can turn off this check. I do see though that it makes some sense to tie this to the site used for the executable and I think the ideal situation is that the OSG 'site' somehow can define this property, but I think at this point in the code, we can't really read site-specific information. 
Unless, we allow that somehow?